app/controllers/reports/regions_controller.rb

```diff
@@ -153,34 +154,37 @@ def set_force_cache
     RequestStore.store[:force_cache] = true if force_cache?
   end
 
-  def region_source
-    @region_source ||= authorize {
-      case region_class
-      when "FacilityDistrict"
-        scope = current_admin.accessible_facilities(:view_reports)
-        FacilityDistrict.new(name: report_params[:id], scope: scope)
-      when "FacilityGroup"
-        current_admin.accessible_facility_groups(:view_reports).find_by!(slug: report_params[:id])
-      when "Block" # we don't have first class auth on Blocks yet, so we authorize via the parent FacilityGroup
-        block = Region.find_by!(slug: report_params[:id], region_type: "block")
-        owning_facility_group = block.district_region.source
-        unless current_admin.accessible_facility_groups(:view_reports).include?(owning_facility_group)
-          raise UserAccess::NotAuthorizedError
-        end
-        block
-      when "Facility"
-        current_admin.accessible_facilities(:view_reports).find_by!(slug: params[:id])
-      else
-        raise ActiveRecord::RecordNotFound, "unknown region_class #{region_class}"
-      end
-    }
-  end
-
   def find_region
-    @region = if current_admin.feature_enabled?(:region_reports)
-      region_source.region
+    @region ||= if current_admin.feature_enabled?(:region_reports)
+      authorize {
+        case region_class
+        when "FacilityDistrict"
+          scope = current_admin.accessible_facilities(:view_reports)
+          FacilityDistrict.new(name: report_params[:id], scope: scope)
```

Comment:
Should this not be `FacilityDistrict.new().region`?

Reply:
The `FacilityDistrict` is a weird one. It doesn't have any backing regions - I think it needs to stay as is, but let me doublecheck its behavior with the flag both off and on.

Reply:
Took it for a detailed spin. The special "Facility DIstrict" links still work great in both cases - and links to these pages are not exposed anywhere on the dashboard 👍